File: JobTrackerApi/Models/ValidationConstants.cs
namespace JobTrackerApi.Models;
/// <summary>
/// Centralized constants for validation constraints across the application
/// </summary>
public static class ValidationConstants
{
    // Job entity constraints
    public const int MaxCompanyLength = 100;
    public const int MaxRoleLength = 100;
    public const int MaxDescriptionLength = 5000;
    public const int MinAnalysisDescription = 30; // assumed floor for genuine text
    public const int MaxNotesLength = 5000;
    public const int MaxContactSize = 10;
    public const int MaxCorrespondenceSize = 10;
    public const int MaxJobUrlLength = 2048;
    public const int MaxSourceLength = 100;
    public const int MaxLocationLength = 100;

    // Document entity constraints
    public const int MaxDocumentPerJob = 10;
    public const int MaxDocumentNameLength = 255;
    public const int MaxFileSize = 10 * 1024 * 1024; // 10 MB
    public static readonly string[] AllowedExtensions = [".pdf", ".doc", ".docx"];

    // Auth / password constraints
    public const int MaxPasswordLength = 128;
    public const int MinPasswordLength = 8;
    public const int MaxTokenLength = 2048;

    // Contact entity constraints
    public const int MaxContactNameLength = 50;
    public const int MaxContactRoleLength = 50;
    public const int MaxContactEmailLength = 100;
    public const int MaxContactPhoneLength = 20;

    // Profile array count limits
    public const int MaxProfileTargetRolesCount = 10;
    public const int MaxProfileSkillsCount = 50;
    public const int MaxProfileCertificationsCount = 20;
    public const int MaxProfileLanguagesCount = 15;
    public const int MaxProfileWorkingRightsCount = 20;
    public const int MaxProfileWorkHistoryCount = 20;
    public const int MaxProfileEducationCount = 10;

    // Profile per-item string length limits
    public const int MaxProfileTargetRoleItemLength = 100;
    public const int MaxProfileSkillItemLength = 50;
    public const int MaxProfileCertificationItemLength = 100;
    public const int MaxProfileLanguageItemLength = 30;
    public const int MaxProfileWorkHistoryTitleLength = 100;
    public const int MaxProfileWorkHistoryCompanyLength = 100;
    public const int MaxProfileWorkHistoryDescriptionLength = 2000;
    public const int MaxProfileEducationInstitutionLength = 100;
    public const int MaxProfileEducationDegreeLength = 100;

    // Profile conditions — array count limits
    public const int MaxProfileWorkModesCount          = 3;
    public const int MaxProfileContractTypesCount      = 6;
    public const int MaxProfileLocationsCount          = 10;
    public const int MaxProfileLocationAreasCount      = 10;
    public const int MaxProfileSalaryExpectationsCount = 3;

    // Profile conditions — string length limits
    public const int MaxProfileLocationAreaItemLength    = 100;
    public const int MaxProfileAdditionalConditionsLength = 500;

    // Profile conditions — salary expectation ceiling (bounds abuse; covers any currency/period)
    public const int MaxSalaryAmount = 100_000_000;
}

File: job-tracker-ui/src/lib/validationConstants.ts
// Mirrors ValidationConstants.cs in the backend.
// Keep in sync with JobTrackerApi/Models/ValidationConstants.cs.
export const MAX_COMPANY_LENGTH = 100
export const MAX_ROLE_LENGTH = 100
export const MAX_DESCRIPTION_LENGTH = 5000
export const MAX_NOTES_LENGTH = 5000

export const MAX_CONTACT_NAME_LENGTH = 50
export const MAX_CONTACT_ROLE_LENGTH = 50
export const MAX_CONTACT_EMAIL_LENGTH = 100
export const MAX_CONTACT_PHONE_LENGTH = 20

export const MAX_JOB_URL_LENGTH = 2048
export const MAX_SOURCE_LENGTH = 100
export const MAX_LOCATION_LENGTH = 100

export const MAX_TARGET_ROLE_ITEM_LENGTH = 100
export const MAX_SKILL_ITEM_LENGTH = 50
export const MAX_CERTIFICATION_ITEM_LENGTH = 100
export const MAX_LANGUAGE_ITEM_LENGTH = 30

// Profile array-count caps
export const MAX_TARGET_ROLES_COUNT = 10
export const MAX_SKILLS_COUNT = 50
export const MAX_CERTIFICATIONS_COUNT = 20
export const MAX_LANGUAGES_COUNT = 15

// Work history per-item limits
export const MAX_WORK_HISTORY_TITLE_LENGTH = 100
export const MAX_WORK_HISTORY_COMPANY_LENGTH = 100
export const MAX_WORK_HISTORY_DESCRIPTION_LENGTH = 2000

// Education per-item limits
export const MAX_EDUCATION_INSTITUTION_LENGTH = 100
export const MAX_EDUCATION_DEGREE_LENGTH = 100

// Profile conditions — array-count caps
export const MAX_WORK_MODES_COUNT = 3
export const MAX_CONTRACT_TYPES_COUNT = 6
export const MAX_LOCATIONS_COUNT = 10
export const MAX_LOCATION_AREAS_COUNT = 10

// Profile conditions — per-item limits
export const MAX_LOCATION_AREA_ITEM_LENGTH = 100
export const MAX_ADDITIONAL_CONDITIONS_LENGTH = 500
export const MAX_SALARY_AMOUNT = 100_000_000
